sdk/checksum/checksum.go

```diff
@@ -26,13 +26,26 @@ func HexChecksum(b []byte) string {
 // Chunk - split bytes to chunk limits
 func Chunk(buf []byte, lim int) [][]byte {
 	var chunk []byte
-	chunks := make([][]byte, 0, len(buf)/lim+1)
+	bufSize := len(buf)
+
+	if bufSize == 0 {
+		return [][]byte{}
+	}
+
+	if bufSize <= lim {
+		return [][]byte{buf}
+	}
+
+	chunks := make([][]byte, 0)
```

Comment:
Why do we no longer set the capacity for `chunks`?

Reply:
Donal answered here at the office, so the reason was that the tool was complaining about our capacity potentially overflowing.

FWIW, I did the math on this one. The capacity was `len(buf)/lim+1` meaning the only way we exceed the maximum length of an integer (i.e. overflow) is if the buffer (=`buf`) had 2^63 elements in it (so 9,223,372,036,854,775,807 elements) on a 64-bit system. Since `buf` is a byte slice, that means you would have 2^63 bytes, which translates to ~9,223 **peta**bytes, meaning you definitely would have run into problem far earlier than when entering this.